### PR TITLE
core: No more UseHttp2

### PR DESCRIPTION
**Pull request by raboof**
Refs #2895

### CODE REVIEW

**Comment by raboof** [comment]
PLS BUILD

File: akka-http-core/src/main/scala/akka/http/scaladsl/ConnectionContext.scala [hunk 1]
```diff
@@ -98,28 +70,9 @@ final class HttpsConnectionContext(
...
-  @deprecated("This method is planned to disappear in 10.2.0", "10.1.9")
-  def this(http2: UseHttp2) = this()
 
-  /** This method is planned to disappear in 10.2.0 */
-  @Deprecated
-  override def withHttp2(newValue: javadsl.UseHttp2): javadsl.HttpConnectionContext = this
-  @deprecated("This method is planned to disappear in 10.2.0", "10.1.9")
-  def withHttp2(newValue: UseHttp2): javadsl.HttpConnectionContext = this
-}
+sealed class HttpConnectionContext extends akka.http.javadsl.HttpConnectionContext with ConnectionContext
```
Comment (johanandren) [inline akka-http-core/src/main/scala/akka/http/scaladsl/ConnectionContext.scala]: `Not for user extension`

File: akka-http-core/src/main/mima-filters/10.1.11.backwards.excludes/2095-no-UseHttp2.backwards.excludes [hunk 2]
```diff
@@ -0,0 +1,22 @@
+ProblemFilters.exclude[MissingClassProblem]("akka.http.scaladsl.UseHttp2*")
```
Comment (jrudolph) [inline akka-http-core/src/main/mima-filters/10.1.11.backwards.excludes/2095-no-UseHttp2.backwards.excludes]: Maybe add a comment that this is removal of deprecated API.

Comment (raboof) [inline akka-http-core/src/main/mima-filters/10.1.11.backwards.excludes/2095-no-UseHttp2.backwards.excludes]:
```suggestion
# Removal of explicitly-deprecated API
ProblemFilters.exclude[MissingClassProblem]("akka.http.scaladsl.UseHttp2*")
```

**Pull request merged**